### PR TITLE
Enable PNG exif orientation test

### PR DESCRIPTION
**Pull request by sturkmen72**
### Pull Request Readiness Checklist
related to #23122

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by sturkmen72** [comment]
RFC : `IMREAD_ADD_ALPHA = 5` this means `IMREAD_COLOR | IMREAD_ANYCOLOR` but i think we can use it.
another options:
`IMREAD_ADD_ALPHA = -2`
`IMREAD_ADD_ALPHA = 512`

File: modules/imgcodecs/src/loadsave.cpp [hunk 1]
```diff
@@ -83,6 +83,9 @@ static Size validateInputImageSize(const Size& size)
 
 static inline int calcType(int type, int flags)
 {
+    if (flags == IMREAD_ADD_ALPHA)
```
Comment (asmorkalov) [inline modules/imgcodecs/src/loadsave.cpp:86]: There is [cv::IMREAD_ANYCOLOR](https://docs.opencv.org/4.x/d8/d6a/group__imgcodecs__flags.html#gga61d9b0126a3e57d9277ac48327799c80ab6573b69300c092b61800222fe555953) option. It should presume alpha channel, if it's present in the original image.

Comment (sturkmen72) [reply modules/imgcodecs/src/loadsave.cpp:86]: see the issue #21695 related cv::IMREAD_ANYCOLOR

File: modules/imgcodecs/test/test_png.cpp [hunk 2]
```diff
@@ -185,6 +185,29 @@ TEST_P(Imgcodecs_PNG_Exif, DISABLED_exif_orientation)
     EXPECT_GE(vec.val[0], colorThresholdHigh);
     EXPECT_LE(vec.val[1], colorThresholdLow);
     EXPECT_LE(vec.val[2], colorThresholdLow);
+
+    m_img = imread(filename, IMREAD_ADD_ALPHA);
```
[the remark this answers is not in the document]
Comment (sturkmen72) [reply modules/imgcodecs/test/test_png.cpp]: when
```
m_img = imread(filename, IMREAD_ANYCOLOR);

    ASSERT_FALSE(m_img.empty());

    EXPECT_EQ(4, m_img.channels());

```
**the test fails **

[----------] 8 tests from ExifFiles/Imgcodecs_PNG_Exif
[ RUN      ] ExifFiles/Imgcodecs_PNG_Exif.exif_orientation/0, where GetParam() = "readwrite/testExifOrientation_1.png"
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(191): error: Expected equality of these values:
  4
  m_img.channels()
    Which is: 3
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(196): error: Expected: (vec4.val[0]) <= (colorThresholdLow), actual: '\xFE' (254) vs 5
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(198): error: Expected: (vec4.val[2]) >= (colorThresholdHigh), actual: '\0' vs 250
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(202): error: Expected: (vec4.val[0]) <= (colorThresholdLow), actual: '\xFF' (255) vs 5
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(203): error: Expected: (vec4.val[1]) >= (colorThresholdHigh), actual: '\0' vs 250
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(208): error: Expected: (vec4.val[0]) >= (colorThresholdHigh), actual: '\x2' (2) vs 250
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(209): error: Expected: (vec4.val[1]) <= (colorThresholdLow), actual: '\xFD' (253) vs 5
[  FAILED  ] ExifFiles/Imgcodecs_PNG_Exif.exif_orientation/0, where GetParam() = "readwrite/testExifOrientation_1.png" (2 ms)
[ RUN      ] ExifFiles/Imgcodecs_PNG_Exif.exif_orientation/1, where GetParam() = "readwrite/testExifOrientation_2.png"
C:\build\precommit_windows64\4.x\opencv\modules\imgcodecs\test\test_png.cpp(191): error: Expected equality of these values:
  4
  m_img.channels()
    Which is: 3
```

**Comment by sturkmen72** [comment]
the name for a new flag maybe cv::IMREAD_UNCHANGED_WITH_ALPHA

**Comment by sturkmen72** [comment]
as stated in the test code
```
 * The flags parameter of the imread function is set as IMREAD_COLOR | IMREAD_ANYCOLOR | IMREAD_ANYDEPTH.
 * Using this combination is an undocumented trick to load images similarly to the IMREAD_UNCHANGED flag,
 * preserving the alpha channel (if present) while also applying the orientation.
```
I think this is acceptable until something better is done.
I can add a new flag, but I think it would be better if it did not appear in the documentation for a while.

File: modules/imgcodecs/test/test_exif.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+ * | R | G |
+ * |-------|
+ * | B | W |
+ * ---------
+ *
+ * Note:
+ * The flags parameter of the imread function is set as IMREAD_COLOR | IMREAD_ANYCOLOR | IMREAD_ANYDEPTH.
+ * Using this combination is an undocumented trick to load images similarly to the IMREAD_UNCHANGED flag,
+ * preserving the alpha channel (if present) while also applying the orientation.
+ */
```
Comment (asmorkalov) [inline modules/imgcodecs/test/test_exif.cpp:52]: I propose to add it to documentation.

Comment (vrabaud) [reply modules/imgcodecs/test/test_exif.cpp:52]: Yes please !

**Pull request merged**